# HG changeset patch
# User Rik <rik@octave.org>
# Date 1517548017 28800
#      Thu Feb 01 21:06:57 2018 -0800
# Node ID 5aa9c885ea18a1a81e44a6423bf0b4a742dc1eb1
# Parent  62a05d23cd0016041de40642c983dc6ed7cf1c08
Rewrite certain Java BIST tests to pass with Java 9 (bug #52627).

* ov-java.cc: Determine whether Java version number is old style (dotted
notation) or new style (single number) before testing.  Tag test with
Integer.compareTo with bug ID 51804.

diff --git a/libinterp/octave-value/ov-java.cc b/libinterp/octave-value/ov-java.cc
--- a/libinterp/octave-value/ov-java.cc
+++ b/libinterp/octave-value/ov-java.cc
@@ -3092,19 +3092,24 @@ equivalent
 
   err_disabled_feature ("javaMethod", "Java");
 
 #endif
 }
 
 /*
 %!testif HAVE_JAVA; usejava ("jvm")
-%! ## Check for valid first two Java version numbers
-%! jver = strsplit (javaMethod ("getProperty", "java.lang.System", "java.version"), ".");
-%! assert (isfinite (str2double (jver{1})) && isfinite (str2double (jver{2})));
+%! jver = javaMethod ("getProperty", "java.lang.System", "java.version");
+%! jver = strsplit (jver, ".");
+%! if (numel (jver) > 1)
+%!   assert (isfinite (str2double (jver{1})));
+%!   assert (isfinite (str2double (jver{2})));
+%! else
+%!   assert (isfinite (str2double (jver{1})));
+%! endif
 */
 
 DEFUN (__java_get__, args, ,
        doc: /* -*- texinfo -*-
 @deftypefn {} {@var{val} =} __java_get__ (@var{obj}, @var{name})
 Get the value of the field @var{name} of the Java object @var{obj}.
 
 For static fields, @var{obj} can be a string representing the fully
@@ -3407,17 +3412,17 @@ Return true if @var{x} is a Java object.
 
 %!testif HAVE_JAVA; usejava ("jvm")
 %! n = javaObject ("java.lang.Long", (int64 (1)));
 %! assert (n.compareTo (int64 (0)), 1);
 %! assert (n.compareTo (int64 (1)), 0);
 %! assert (n.compareTo (int64 (2)), -1);
 %! assert (n.toString (), "1");
 
-%!testif HAVE_JAVA; usejava ("jvm")
+%!testif HAVE_JAVA; usejava ("jvm") <51804>
 %! n = javaObject ("java.lang.Integer", 1.35);
 %! assert (n.compareTo (0), 1);
 %! assert (n.compareTo (1), 0);
 %! assert (n.compareTo (2), -1);
 
 %!testif HAVE_JAVA; usejava ("jvm")
 %! n = javaObject ("java.lang.Short", 1.35);
 %! assert (n.compareTo (0), 1);
